File: src/dynamicTyping/python/duck.py
class Mallard:
  def dance(self):
    return " _/¯ "
  def talk(self):
    return "quack quack"

class Wolf:
  def eat(self, duck):
    return " 😈 "
  def dance(self):
    return " ¯\\_()_/¯ "
  def talk(self):
    return "QUACK QUACK WHOO"

class Forest:
  def twoDucksAlone(self, d1, d2):
      print(d1.dance())
      print(d2.dance())
      print(d1.talk())
      print(d2.talk())

wolf = Wolf()
theDuck = wolf
aDuck = Mallard()
aForest = Forest()


aForest.twoDucksAlone(aDuck, theDuck)
print(wolf.eat(aDuck))

File: src/dynamicTyping/ruby/duck.rb
class Mallard
    def dance()
        " _/¯ "
    end

    def talk()
        "quack quack"
    end
end


class Wolf

    def eat(duck)
        " 😈 "
    end

    def dance()
        " ¯\\_()_/¯ "
    end

    def talk()
        "QUACK QUACK WHOO"
    end
end

class Forest
    def twoDucksAlone(d1, d2)
        puts d1.dance
        puts d2.dance
        puts d1.talk
        puts d2.talk
    end
end

wolf = Wolf.new
theDuck = wolf
aDuck = Mallard.new
aForest = Forest.new

aForest.twoDucksAlone(aDuck, theDuck)
puts wolf.eat(aDuck)
